app/tests/spec/models/flow.js

```diff
@@ -0,0 +1,205 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var assert = require('chai').assert;
+  var AuthErrors = require('lib/auth-errors');
+  var Flow = require('models/flow');
+  var ResumeToken = require('models/resume-token');
+  var sinon = require('sinon');
+  var Url = require('lib/url');
+  var WindowMock = require('../../mocks/window');
+
+  var FLOW_ID = 'F1031DF1031DF1031DF1031DF1031DF1031DF1031DF1031DF1031DF1031DF103';
+  var FLOW_ID2 = '71031D71031D71031D71031D71031D71031D71031D71031D71031D71031D7103';
+
+  describe('models/flow', function () {
+    var flow;
+    var sentryMetricsMock;
+    var windowMock;
+
+    function createFlow() {
+      flow = new Flow({
+        sentryMetrics: sentryMetricsMock,
+        window: windowMock
+      });
+    }
+
+    beforeEach(function () {
+      $('body').removeData('flowId').removeAttr('data-flow-id');
+      $('body').removeData('flowBegin').removeAttr('data-flow-begin');
+      sentryMetricsMock = {
+        captureException: sinon.spy()
+      };
+      windowMock = new WindowMock();
+    });
+
+    afterEach(function () {
+      flow = null;
+    });
+
+    it('fetches from the `resume` search parameter, if available', function () {
+      windowMock.location.search = Url.objToSearchString({
+        resume: ResumeToken.stringify({ flowBegin: 42, flowId: FLOW_ID })
+      });
+
+      createFlow();
+
+      assert.equal(flow.get('flowId'), FLOW_ID);
+      assert.equal(flow.get('flowBegin'), 42);
+    });
+
+    it('fetches from body data attributes, if available', function () {
+      windowMock.$('body').attr('data-flow-id', FLOW_ID);
```

Comment:
I think I was trying to use a mock `<body>` element for some of these tests, but it didn't really work out.  Your cleanup sounds good.